CharlieBackend.Api.UnitTes/DashboardServiceTest.cs

```diff
@@ -0,0 +1,535 @@
+            Assert.Empty(resultWithoutStudent.Data.AverageStudentsMarks);
+            Assert.Empty(resultWithoutStudent.Data.AverageStudentVisits);
+        }
+
+        [Fact]
+        public async Task GetStudentClassbook()
+        {
+            //Arrange
+            long studentIdWithGroup = 7;
+            long studentIdWithoutGroup = 20;
+            
+            var claims = new List<Claim>()
+            {
+                new Claim(ClaimTypes.Name, "Denys"),
+                new Claim(ClaimTypes.NameIdentifier , "1"),
+                new Claim(ClaimTypes.Role, UserRole.Student.ToString()),
+                new Claim("Id", "7")
+            };
+
+            var claimIdentity = new ClaimsIdentity(claims, "TestAuthType");
+            var claimPrincipal = new ClaimsPrincipal(claimIdentity);
+
+            var claimsForStudentWitoutGroup = new List<Claim>()
+            {
+                new Claim(ClaimTypes.Name, "Denys"),
+                new Claim(ClaimTypes.NameIdentifier , "1"),
+                new Claim(ClaimTypes.Role, UserRole.Student.ToString()),
+                new Claim("Id", "20")
+            };
+
+            var claimIdentityForStudentWitoutGroup = new ClaimsIdentity(claimsForStudentWitoutGroup, "TestAuthType");
+            var claimPrincipalForStudentWitoutGroup = new ClaimsPrincipal(claimIdentityForStudentWitoutGroup);
+
+            var mockPrincipal = new Mock<IPrincipal>();
+            mockPrincipal.Setup(x => x.Identity).Returns(claimIdentity);
+            mockPrincipal.Setup(x => x.Identity).Returns(claimIdentityForStudentWitoutGroup);
+            mockPrincipal.Setup(x => x.IsInRole(UserRole.Student.ToString())).Returns(true);
+            mockPrincipal.Setup(x => x.IsInRole(UserRole.Mentor.ToString())).Returns(false);
+
+            var dashbordAnaliticRequstWithData = new DashboardAnalyticsRequestDto<ClassbookResultType>()
+            {
+                StartDate = new DateTime(2011, 1, 1),
+                FinishDate = new DateTime(2017, 5, 20),
+                IncludeAnalytics = new ClassbookResultType []
+                {
+                    ClassbookResultType.StudentMarks,
+                    ClassbookResultType.StudentPresence
+                }
+            };
+
+            var dashbordAnaliticRequstWithoutClassbook = new DashboardAnalyticsRequestDto<ClassbookResultType>()
+            {
+                StartDate = new DateTime(2011, 1, 1),
+                FinishDate = new DateTime(2017, 5, 20),
+            };
+
+            var dashbordRepositoryMock = new Mock<IDashboardRepository>();
+
+            dashbordRepositoryMock.Setup(x => x.GetGroupsIdsByStudentIdAndPeriodAsync(
+                studentIdWithGroup, dashbordAnaliticRequstWithData.StartDate, dashbordAnaliticRequstWithData.FinishDate))
+                .ReturnsAsync(new List<long> { 1 });
+            
+            dashbordRepositoryMock.Setup(x => x.GetGroupsIdsByStudentIdAndPeriodAsync(
```

Comment:
in this method you rewrite action of line 279. So or you have to give not "studentIdWithoutGroup" argument here. Because for same arguments method can not have to 2 different mock actions